presto-docs/src/main/sphinx/functions/aggregate.rst

```diff
@@ -564,8 +564,164 @@ To find the `ROC curve <https://en.wikipedia.org/wiki/Receiver_operating_charact
+
+    Returns the approximate log-2 differential entropy from a random variable's sample outcomes. The function
+    internally creates a weighted reservoir (see [Efraimidis2006]_), then calculates the
+    entropy from the sample results by approximating the derivative of the cumulative distribution
+    (see [Alizadeh2010]_).
+
+    ``sample_size`` is the maximal number of reservoir samples.
+
+    ``x`` (``double``) is the samples.
+
+    ``weight`` (``double``) is a non-negative double value indicating the weight of the sample.
+
+    For example, to find the differential entropy of ``x`` with weights ``weight`` of ``data``
+    using 1000000 reservoir samples, use
+
+    .. code-block:: none
+
+         SELECT
+             differential_entropy(1000000, x, weight)
+         FROM
+             data
+
+    .. note::
+
+        If :math:`x` has a known lower and upper bound,
+        prefer the versions taking ``(bucket_count, x, weight, "fixed_histogram_mle", min, max)``,
+        or ``(bucket_count, x, weight, "fixed_histogram_jacknife", min, max)``,
+        as they have better convergence.
+
+.. function:: differential_entropy(bucket_count, x, weight, "fixed_histogram_mle", min, max) -> double
+
+    Returns the approximate log-2 differential entropy from a random variable's sample outcomes. The function
+    internally creates a conceptual histogram of the sample values, calculates the counts, and
+    then approximates the entropy using maximum-likelihood estimation (see [Beirlant2001]_).
+
+    ``bucket_count`` (``long``) is the number of histogram buckets.
+
+    ``x`` (``double``) is the samples.
+
+    ``min`` and ``max`` are double values indicating minimum and maximum values of the samples. This function
+    raises an error if there is a sample outside of this range.
+
+    ``weight`` is a non-negative double value indicating the weight of the sample.
+
+    For example, to find the differential entropy of ``x``, each between ``0.0`` and ``1.0``,
+    with weights ``weight`` of ``data``
+    using 1000000 reservoir samples, use
+
+    .. code-block:: none
+
+         SELECT
+             differential_entropy(1000000, x, weight, 'fixed_histogram_mle', 0.0, 1.0)
+         FROM
+             data
+
+    .. note::
+
+        If :math:`x` doesn't have known lower and upper bounds, prefer the versions taking ``(sample_size, x)``
+        (unweighted) of ``(sample_size, x, weight)`` (weighted), as they use reservoir sampling which doesn't require
+        a known range for samples. Otherwise, if the number of distinct weights is low,
+        especially if the number of samples is low, consider using the version taking
+        ``(bucket_count, x, weight, "fixed_histogram_jacknife", min, max)``, as this uses jacknife bias correction,
+        which is better than maximum likelihood estimation.
+
+.. function:: differential_entropy(bucket_count, x, weight, "fixed_histogram_jacknife", min, max) -> double
```

Comment:
How would you feel about merging the documentation of this function with the previous one? There is a lot of copy-paste and the only difference is the value of the method parameter. The documentation of the method parameter can contain instructions on when to use which value.

`function:: differential_entropy(bucket_count, x, weight, method, min, max) -> double`